temporian/core/operators/test/test_tick.py

```diff
@@ -72,6 +72,94 @@ def test_align_single(self):
+        expected_output = event_set([1, 5, 9])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+        # Not aligned, no include right, end doesn't match last tick
+        result = evset.tick(3.0, align=False, include_right=False)
+        expected_output = event_set([1, 4, 7])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+
+        # Aligned, no include right, end doesn't match last tick
+        result = evset.tick(4.0, align=True, include_right=False)
+        expected_output = event_set([4, 8])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+        # Not aligned, no include right, end matches last tick
+        result = evset.tick(3.0, align=True, include_right=False)
+        expected_output = event_set([3, 6, 9])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+
+    def test_inclusive_right(self):
+        evset = event_set([1, 5, 9])
+        # Not aligned, include right, end matches last tick
+        result = evset.tick(4.0, align=False, include_right=True)
+        expected_output = event_set([1, 5, 9])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+        # Not aligned, include right, end doesn't match last tick
+        result = evset.tick(3.0, align=False, include_right=True)
+        expected_output = event_set([1, 4, 7, 10])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+
+        # Aligned, include right, end doesn't match last tick
+        result = evset.tick(4.0, align=True, include_right=True)
+        expected_output = event_set([4, 8, 12])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+        # Not aligned, include right, end matches last tick
+        result = evset.tick(3.0, align=True, include_right=True)
+        expected_output = event_set([3, 6, 9])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+
+    def test_not_inclusive_left(self):
+        evset = event_set([1, 5, 9])
+        # Not aligned, no include left, start matches first tick
+        result = evset.tick(4.0, align=False, include_left=False)
+        expected_output = event_set([1, 5, 9])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+
+        # Aligned, no include left, start doesn't match first tick
+        result = evset.tick(4.0, align=True, include_left=False)
+        expected_output = event_set([4, 8, 12])
+        assertOperatorResult(
+            self, result, expected_output, check_sampling=False
+        )
+
+    def test_inclusive_left(self):
```

Comment:
Can you also try with negative timestamps?

Reply:
that's a good catch! I totally forgot about negative timestamps